Tests/CocoaLumberjackTests/DDContextFilterLogFormatterTests.m

```diff
@@ -0,0 +1,176 @@
+//
+// * Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+//
+// * Neither the name of Deusty nor the names of its contributors may be used
+//   to endorse or promote products derived from this software without specific
+//   prior written permission of Deusty, LLC.
+
+#import <XCTest/XCTest.h>
+#import <CocoaLumberjack/DDContextFilterLogFormatter.h>
+
+static DDLogMessage *testLogMessage() {
+    return [[DDLogMessage alloc] initWithMessage:@"test log message"
+                                           level:DDLogLevelDebug
+                                            flag:DDLogFlagError
+                                         context:1
+                                            file:@(__FILE__)
+                                        function:@(__func__)
+                                            line:__LINE__
+                                             tag:NULL
+                                         options:(DDLogMessageOptions)0
+                                       timestamp:nil];
+}
+
+@interface DDContextAllowlistFilterLogFormatterTests : XCTestCase
+@property (nonatomic, strong, readwrite) DDContextAllowlistFilterLogFormatter *filterLogFormatter;
+@end
+
+@implementation DDContextAllowlistFilterLogFormatterTests
+
+- (void)setUp {
+    [super setUp];
+    self.filterLogFormatter = [[DDContextAllowlistFilterLogFormatter alloc] init];
+}
+
+- (void)tearDown {
+    self.filterLogFormatter = nil;
+    [super tearDown];
+}
+
+- (void)testAllowlistFilterLogFormatter {
+    XCTAssertEqualObjects([self.filterLogFormatter allowlist], @[]);
+    XCTAssertFalse([self.filterLogFormatter isOnAllowlist:1]);
+    XCTAssertNil([self.filterLogFormatter formatLogMessage:testLogMessage()]);
+    
+    [self.filterLogFormatter addToAllowlist:1];
+    XCTAssertEqualObjects([self.filterLogFormatter allowlist], @[@1]);
+    XCTAssertTrue([self.filterLogFormatter isOnAllowlist:1]);
+    XCTAssertEqualObjects([self.filterLogFormatter formatLogMessage:testLogMessage()], @"test log message");
+    
+    [self.filterLogFormatter removeFromAllowlist:1];
+    XCTAssertEqualObjects([self.filterLogFormatter allowlist], @[]);
+    XCTAssertFalse([self.filterLogFormatter isOnAllowlist:1]);
+    XCTAssertNil([self.filterLogFormatter formatLogMessage:testLogMessage()]);
+}
+
+@end
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+@interface DDContextWhitelistFilterLogFormatterTests : XCTestCase
+@property (nonatomic, strong, readwrite) DDContextWhitelistFilterLogFormatter *filterLogFormatter;
+@end
+
+@implementation DDContextWhitelistFilterLogFormatterTests
+
+- (void)setUp {
+    [super setUp];
+    self.filterLogFormatter = [[DDContextWhitelistFilterLogFormatter alloc] init];
+}
+
+- (void)tearDown {
+    self.filterLogFormatter = nil;
+    [super tearDown];
+}
+
+- (void)testWhitelistFilterLogFormatter {
```

Comment:
Also here, we could have one test suite with two methods.
One method for new "allow" methods.
Another for old "whitelist" methods.